bdns/dns.go

```diff
@@ -151,13 +150,32 @@ var (
 	}
 )
 
-type ResolverAddr string
+// ResolverAddrs contains DNS resolver(s) that were chosen to perform a
+// validation request or CAA recheck. A ResolverAddr will be in the form of
+// host:port, A:host:port, or AAAA:host:port depending on which type of lookup
+// was done.
+type ResolverAddrs []string
+
+// Append appends a variadic amount of non-zero value strings to a slice of
+// ResolverAddrs. Zero value strings are discard.
+func (r *ResolverAddrs) append(s ...string) {
+	fmt.Printf("pre-loop length: %d\n", len(*r))
+
+	for _, element := range s {
+		if element == "" {
+			continue
+		}
+		*r = append(*r, element)
+		fmt.Printf("appended %+v\n", element)
+	}
+	fmt.Printf("post-loop length: %d\n", len(*r))
+}
```

Comment:
This can be written a little more compactly as well.

```suggestion
func (r *ResolverAddrs) append(addresses ...string) {
	for _, addr := range addresses {
		if addr != "" {
			*r = append(*r, addr)
		}
	}
}
```